### PR TITLE
Make checksum comparison try matching with lf and crlf line endings

### PR DESCRIPTION
**Pull request by tomhoule**
See code for documentation on the how and why.

Relevant issues:

- https://github.com/prisma/prisma/issues/7398
- https://github.com/prisma/prisma/issues/7101

### CODE REVIEW

File: migration-engine/connectors/migration-connector/src/migrations_directory.rs [hunk 1]
```diff
@@ -201,40 +200,11 @@ impl MigrationDirectory {
...
-        let script = self.read_migration_script()?;
-        let mut hasher = Sha512::new();
-        hasher.update(&script);
-        let bytes = hasher.finalize();
-
-        buf.clear();
-        buf.extend_from_slice(bytes.as_ref());
-
-        Ok(())
-    }
```
Comment (tomhoule) [inline migration-engine/connectors/migration-connector/src/migrations_directory.rs:215]: This was dead code :(

**Comment by Jolg42** [comment]
From Tom: will need a careful review but I think it's complete.

**Review by do4gr — APPROVED** [review]
LGTM

**Review by pimeys — APPROVED** [review]
for loops kind of caused a bit of a friction when reading this, but I can approve this!

**Pull request merged**